components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -200,6 +200,7 @@ interface ERCTokenRegistry {
   GetTokenBySymbol(string symbol) => (ERCToken? token);
   GetAllTokens() => (array<ERCToken> tokens);
   GetBuyTokens() => (array<ERCToken> tokens);
+  GetBuyUrl(string address, string symbol, string amount) => (string url);
```

Comment:
I created a separate issue to switch to that method on desktop https://github.com/brave/brave-browser/issues/18377. As currently it's hardcoded inside ts https://github.com/brave/brave-core/blob/master/components/brave_wallet_ui/utils/buy-asset-url.ts#L9